### PR TITLE
Support kinesis as event stream

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

Support kinesis as event stream

### CODE REVIEW

File: builds/jazz-build-module/aws-lambda-events-module.groovy [hunk 1]
```diff
@@ -14,6 +14,83 @@ def initialize(configLoader){
...
+def checkKinesisAndAddLambdaTrigger(stream_name, lambdaARN) {
+  try {
+    sh "aws kinesis describe-stream --stream-name ${stream_name} --profile cloud-api --output json"
+    echo "Stream exists and have access"
+    addLambdaTriggerToKinesisStream(true, stream_name, lambdaARN)
+  } catch (ex) {
+    def response
+    try {
+      response = sh(
+        script: "aws kinesis describe-stream --stream-name ${stream_name} --profile cloud-api --output json 2<&1 | grep -c 'ResourceNotFoundException'",
```
Comment (svidhani) [inline builds/jazz-build-module/aws-lambda-events-module.groovy]: did you also add this in SQS?

File: builds/jazz-build-module/aws-lambda-events-module.groovy [hunk 2]
```diff
@@ -14,6 +14,83 @@ def initialize(configLoader){
...
+}
+
+def createKinesisStream(stream_name, lambdaARN){
+  try{
+    sh "aws kinesis create-stream --stream-name ${stream_name} --shard-count 1 --profile cloud-api --output json"
+    def event_source_arn = "arn:aws:kinesis:${config_loader.AWS.REGION}:${config_loader.AWS.ACCOUNTID}:stream/${stream_name}"
+    createEventSourceMapping(event_source_arn, lambdaARN, true)
+  }catch(ex){
+    echo "Failed to create the queue"
+    error "Failed to create the queue"
```
Comment (svidhani) [inline builds/jazz-build-module/aws-lambda-events-module.groovy]: queue or stream?

File: builds/jazz-build-module/aws-lambda-events-module.groovy [hunk 3]
```diff
@@ -14,6 +14,83 @@ def initialize(configLoader){
...
+    echo "Stream exists and have access"
+    addLambdaTriggerToKinesisStream(true, stream_name, lambdaARN)
+  } catch (ex) {
+    def response
+    try {
+      response = sh(
+        script: "aws kinesis describe-stream --stream-name ${stream_name} --profile cloud-api --output json 2<&1 | grep -c 'ResourceNotFoundException'",
+        returnStdout: true
+      ).trim()
+    } catch (e) {
```
[the remark this answers is not in the document]
Comment (svidhani) [reply builds/jazz-build-module/aws-lambda-events-module.groovy]: what if stream exists but you cannot access it, meaning somebody else has the it with same name?

[the remark this answers is not in the document]
Comment (SiniWilson) [reply builds/jazz-build-module/aws-lambda-events-module.groovy]: Logged the error. As per the AWS doc, stream name is scoped by AWS Region.

File: builds/jazz-build-module/aws-lambda-events-module.groovy (hunk 1 above)
Comment (SiniWilson) [reply builds/jazz-build-module/aws-lambda-events-module.groovy]: Yes. Already did in SQS

File: builds/jazz-build-module/aws-lambda-events-module.groovy (hunk 2 above)
Comment (SiniWilson) [reply builds/jazz-build-module/aws-lambda-events-module.groovy]: done

File: builds/jazz-build-module/aws-lambda-events-module.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (SiniWilson) [reply builds/jazz-build-module/aws-lambda-events-module.groovy]: I am taking only the resource not exists error, all  other error will throw error

**Pull request merged**